plugins/azure/resoto_plugin_azure/resource/compute.py

```diff
@@ -2446,8 +2478,18 @@ class AzureVirtualMachineIdentity:
     user_assigned_identities: Optional[Dict[str, AzurePrincipalidClientid]] = field(default=None, metadata={'description': 'The list of user identities associated with the virtual machine. The user identity dictionary key references will be arm resource ids in the form: /subscriptions/{subscriptionid}/resourcegroups/{resourcegroupname}/providers/microsoft. Managedidentity/userassignedidentities/{identityname}.'})  # fmt: skip
 
 
+InstanceStatusMapping = {
+    "Creating": InstanceStatus.BUSY,
+    "Updating": InstanceStatus.BUSY,
+    "Succeeded": InstanceStatus.RUNNING,
+    "Failed": InstanceStatus.STOPPED,
+    "Deleting": InstanceStatus.BUSY,
+    "Migrating": InstanceStatus.BUSY,
```

Comment:
The API itself only defines a string and sadly no enumeration.
According to ChatGPT, those are the possible states:
```
Creating: Indicates that the VM is currently being created.
Updating: Indicates that an existing VM is being updated.
Succeeded: Indicates that the last create, update, or delete operation was successful.
Failed: Indicates that the last create, update, or delete operation failed and provides an error message for debugging.
Deleting: Indicates that the VM is in the process of being removed.
Migrating: Indicates that the VM is being moved to a different host server, resource group, or subscription.
Deallocating: Indicates that the VM is being deallocated, which releases compute resources.
Deallocated: Indicates that the VM is deallocated and not consuming compute resources.
Starting: Indicates that the VM is in the process of being started.
Running: Indicates that the VM is currently running.
Stopping: Indicates that the VM is in the process of being stopped.
Stopped: Indicates that the VM is stopped, but still allocated and incurring compute charges.
```

Reply:
You marked it as resolved - but I do not see the values mapped?

Reply:
I found information only about 6 [providing states](https://learn.microsoft.com/en-us/azure/virtual-machines/states-billing).

<div class="heading-wrapper" data-heading-level="h2"><h2 id="provisioning-states" class="heading-anchor">Provisioning states</h2></div>
<p>The provisioning state is the status of a user-initiated, 
control-plane operation on the VM. These states are separate from the 
power state of a VM.</p>
<div class="has-inner-focus">

Provisioning state | Description
-- | --
Creating | Virtual machine is being created.
Updating | Virtual machine is updating to the latest model. Some non-model  changes to a virtual machine such as start and restart fall under the  updating state.
Failed | Last operation on the virtual machine resource was unsuccessful.
Succeeded | Last operation on the virtual machine resource was successful.
Deleting | Virtual machine is being deleted.
Migrating | Seen when migrating from Azure Service Manager to Azure Resource Manager.

</div>


According to`States and billing status of Azure Virtual Machines`, i decide to do not add other states into `InstanceStatusMapping`, because it intended to [PowerState](https://learn.microsoft.com/en-us/azure/virtual-machines/states-billing#power-states-and-billing).